File: node/greeter/greeter.js
const pglet = require("pglet");

(async () => {
    let p = await pglet.page("greeter");

    let txt_name = await p.send("add textbox label='Your name' description='Please provide your full name'");
    let btn_hello = await p.send("add button primary text='Say hello'");
    
    while(true) {
        const e = await p.waitEvent();
        if (e.target === btn_hello && e.name === 'click') {
            let name = await p.send(`get ${txt_name} value`);
            await p.send("clean page");
            await p.send(`add text value='Hello, ${name}!'`);
        }
    }
})();

File: python/greeter/greeter.py
import sys
import pglet
from pglet import Textbox, Button, Text

p = pglet.page("greeter")
p.clean()

def say_hello_click(e):
    name = p.get_value(txt_name)
    p.clean()
    p.add(Text(value=f'Hello, {name}!'))
    sys.exit()

txt_name = p.add(Textbox(label="Your name", description="Please provide your full name"), Textbox(label="Your name", description="Please provide your full name"))
print("txt_name ", txt_name)
p.add(Button(text="Say hello", primary=True, onclick=say_hello_click))

# wait until browser window is closed or page reloaded
p.wait_close()